internal/state/validation.go

```diff
@@ -122,8 +122,8 @@ func validateBlock(state State, block *types.Block) error {
 
 	case block.Height == state.InitialHeight:
 		genesisTime := state.LastBlockTime
-		if !block.Time.Equal(genesisTime) {
-			return fmt.Errorf("block time %v is not equal to genesis time %v",
+		if block.Time.Before(genesisTime) {
```

Comment:
This case is good, because even with BFT Time we should have `block.Time == genesisTime`.